src/components/table/index.js

```diff
@@ -0,0 +1,193 @@
+    const newobtaineddataSource = [...obtaineddataSource];
+    newobtaineddataSource.sort((a, b) => {
+      let keyA = a[key];
+      let keyB = b[key];
+      if (typeof a[key] === 'string') {
+        keyA = keyA.toLowerCase();
+        keyB = keyB.toLowerCase();
+      }
+      if (keyA < keyB) return -1;
+      if (keyA > keyB) return 1;
+      return 0;
+    });
+    setobtaineddataSource(newobtaineddataSource);
+    const newSortIndex = [...colSortIndex];
+    newSortIndex[index] = true;
+    setSortIndex(newSortIndex);
+  };
+  const sortDataDescending = (key, index) => {
+    const newobtaineddataSource = [...obtaineddataSource];
+    newobtaineddataSource.sort((a, b) => {
+      let keyA = a[key];
+      let keyB = b[key];
+      if (typeof a[key] === 'string') {
+        keyA = keyA.toLowerCase();
+        keyB = keyB.toLowerCase();
+      }
+      if (keyA > keyB) return -1;
+      if (keyA < keyB) return 1;
+      return 0;
+    });
+    setobtaineddataSource(newobtaineddataSource);
+    const newSortIndex = [...colSortIndex];
+    newSortIndex[index] = false;
+    setSortIndex(newSortIndex);
+  };
+
+  useEffect(() => {
+    const newColSortIndex = colNames.map((colName) => {
+      if (colName.sortable) {
+        return true;
+      }
+      return null;
+    });
+    setSortIndex(newColSortIndex);
+  }, []);
+
+  return (
+    <TableContainer {...rest} variant={variant}>
+      <DefaultTable variant={variant}>
+        <thead>
+          <Tr hoverable={hoverable} bordered={bordered} variant={variant}>
+            {colNames.map((colName, index) => (
+              <Th
+                key={colName.key}
+                variant={variant}
+                headerColor={headerColor}
+                style={{ cursor: colName.sortable ? 'pointer' : 'default' }}
+                onClick={
+                  colName.sortable
+                    ? colSortIndex[index]
+                      ? () => sortDataDescending(colName.key, index)
+                      : () => sortDataAscending(colName.key, index)
+                    : null
+                }
+              >
+                {colName.sortable ? (
+                  <HeaderContainer>
+                    <HeaderText variant={variant}>{colName.title}</HeaderText>
+
+                    <ArrowIcon
+                      style={{
+                        transform: colSortIndex[index]
+                          ? 'rotate(0)'
+                          : 'rotate(-180deg)',
+                      }}
+                      variant={variant}
+                    >
+                      {sortIcon}
+                    </ArrowIcon>
+                  </HeaderContainer>
+                ) : (
+                  colName.title
+                )}
+              </Th>
+            ))}
+          </Tr>
+        </thead>
+        <tbody>
+          {obtaineddataSource.map((data) => {
+            return (
+              <Tr
+                key={Math.random()}
```

Comment:
Keys should not be assigned random values.

Reply:
Index is used as keys except of random values.